watertap/tools/oli/watertap_to_oli_helper_functions.py

```diff
@@ -0,0 +1,205 @@
+)
+
+
+def watertap_to_oli(watertap_name: str) -> OLIName:
+    """
+    This method creates a named tuple
+    which can be passed directly into OLI
+    or into MCAS property models.
+
+    Parameters
+    ----------
+    watertap_name : str
+        a substance name in WaterTAP format (typical IDAES convention, i.e., B[OH]4_-)
+
+    Raises
+    ------
+    RuntimeError: this generic error is raised if:
+        - watertap_name is a blank string
+        - an extracted charge_sign is not plus or minus
+        - if more than 1 "_" is present in watertap_name
+
+    Returns
+    -------
+    OLIName: named tuple containing:
+        - oli_name: substance name converted to OLI convention
+        - watertap_name: original name passed into method
+        - charge: charge value
+        - charge_group: categorization under one of: Cations, Anions, Neutrals
+        - molar_mass: weight of 1 mole of the substance (grams per mole)
+
+    """
+    c = findall(r"[A-Z]", watertap_name)
+    print(c)
+    if len(c) < 1:
+        raise RuntimeError(
+            f" At least 1 uppercase letter is required to specify a molecule, not '{watertap_name}'."
+        )
+
+    components = watertap_name.split("_")
+
+    print(components)
+    # neutral molecule
+    if len(components) == 1:
+        molecule = components[0]
+        charge = 0
+
+    # charged molecule
+    elif len(components) == 2:
+        molecule = components[0] + "ION"
+
+        charge = components[1]
+        charge_sign = charge[-1]
+
+        if len(charge) > 1:
+            charge_magnitude = int(charge[:-1])
+        else:
+            charge_magnitude = 1
+
+        if charge_sign == "+":
+            charge = charge_magnitude
+        elif charge_sign == "-":
+            charge = -charge_magnitude
+        else:
+            raise RuntimeError(" Only + and - are valid charge indicators.")
+
+    else:
+        raise RuntimeError(" Unrecognized name format " + watertap_name)
+
+    charge_group = _charge_group_computation(charge)
+
+    oli_name = molecule.replace("[", "").replace("]", "").upper()
+    molar_mass = _molar_mass_from_watertap_name(components[0])
+
+    return OLIName(oli_name, watertap_name, charge, charge_group, molar_mass)
+
+
+def _molar_mass_from_watertap_name(watertap_formula: str) -> float:
+    """
+    This function extracts atomic weight data from a periodic table file
+    to generate the molar mass of a chemical substance.
+
+    This function requires additional testing for complex solutes
+    such as CH3CO2H, [UO2]2[OH]4, etc.
+    """
+
+    file_path = Path(__file__).parents[0]
+    periodic_table = read_csv(join(file_path, "periodic_table.csv"))
+
+    # isolate single- and double- letter elements and add to element_counts dict
+    elements = findall("[A-Z][a-z]?[0-9]*", watertap_formula)
+
+    # TODO: strengthen element criteria to reduce potential for input bugs
+    element_counts = {}
```

Comment:
So I used a package before to do this, e.g. get moleular mass for a molecule called molmass https://pypi.org/project/molmass/  it might be a good option to use instead of manually doing these calculations. a suggestion. 

It might be also be possible to replace our own data base with this library instead.

Reply:
Whoa- `molmass `looks very interesting.  We can definitely get charge and molecular weight from that as well, and it seems rather flexible.

Reply:
I've made a note to look into molmass, it appears to be a more robust solution, but would people have to start renaming everything in molmass convention?